### ISSUE TITLE
`OUTCOME_TRY` does not preserve the `exception_ptr`

### ISSUE THREAD

**Issue by akrzemi1** [ned14/outcome#65]
The following program assertion-fails; as though the `exception_ptr` were not correctly propagated through macro `OUTCOME_TRY`.

```c++
#include "outcome.hpp"
#include <cassert>

namespace outcome = OUTCOME_V2_NAMESPACE;
    
auto f() noexcept -> outcome::outcome<int> 
try {
    throw std::runtime_error{"XXX"};
}
catch (...) {
    return std::current_exception();
}
     
auto g() noexcept -> outcome::outcome<int> 
{
  OUTCOME_TRY (ans, f());
  return ans;
}

int main()
{
    outcome::outcome<int> o = g();
    assert (o.has_exception()); // fails
    // this would succeed: assert (o.has_error());
}
```

Object `o` inside `main()` has error but does not have an excption.

See Wandbox life example here: https://wandbox.org/permlink/gnBqH1vrOwu9ygmt

**Comment by ned14** [comment]
I converted your repro into a test and it does not fail on MSVC. Travis is still cycling the tests for it now, but if the commit github references above goes green, then it's the same on POSIX.

I also replaced your copy of outcome.hpp on wandbox with the latest single header include file from develop branch here and it passes: https://wandbox.org/permlink/2how9WFiTG1MEFAc.

It is possible it got broke temporarily and then got fixed. Though I don't remember making any changes to that code recently. All the commits made recently have been for docs.

Let me know if things are fixed for you too.
